### ISSUE TITLE
embedded queries

### ISSUE THREAD

**Issue by oopcode** [Masterminds/squirrel#128]
Hi! I have no idea how to write an embedded query like this using `squirrel` for PostgreSQL:
```
SELECT * FROM Foo AS f WHERE f.X = 42 AND NOT EXISTS (SELECT * FROM Bar AS b WHERE f.Field = b.AnotherField AND b.Y = 24);
```
Doing this like:
```
outerQuery := squirrel.StatementBuilder.PlaceholderFormat(squirrel.Dollar).Select("*").From("Foo AS f").Where("f.X = ?", 42)
innerQuery := squirrel.StatementBuilder.PlaceholderFormat(squirrel.Dollar).Select("*").Prefix("NOT EXISTS (").Suffix(")").From("Blacklists AS b").Where("f.Field = b.AnotherField").Where("b.Y = ?", 24)
outerQuery = outerQuery.Where(innerQuery)
```

results in an invalid query for PostgreSQL because inner query arguments are numbered independently (starting from `$1`), i.e.:

```
SELECT * FROM Foo AS f WHERE f.X = $1 AND NOT EXISTS (SELECT * FROM Bar AS b WHERE f.Field = b.AnotherField AND b.Y = $1);
```

Do I miss something?

**Comment by lann** [comment]
No, that is definitely a bad behavior. As a workaround, I think if you set `PlaceholderFormat(squirrel.Question)` on the innerQuery it will work; the outerQuery will correct the placeholders.

**Comment by oopcode** [comment]
should I consider writing a fix for this case?

**Comment by lann** [comment]
Sure! This is probably where you would look: https://github.com/Masterminds/squirrel/blob/caff2d522d550f0998629c85898ad35e8a603328/part.go#L21

Off the top of my head, maybe add a new e.g. `rawToSql` method to the statement builders that skips the placeholder replacing?

**Comment by oopcode** [comment]
Yep, that might work! I guess it won’t take me too long.

**Comment by lann** [comment]
Fixed by #129
